### ISSUE TITLE
`AWS` vs `Aws` - uninitialized constant Paperclip::Storage::S3::AWS

### ISSUE THREAD

**Issue by professor** [thoughtbot/paperclip#2128]
We're trying to install Paperclip with S3 on a new Rails project, and we ran into version compatibility troubles with the `AWS` constant not being found when using Paperclip 4.3.5 and aws-sdk 2.2.25.

The documentation seems to indicate that the current version of Paperclip is compatible with the current version of aws-sdk. 

A solution to this was to downgrade to aws-sdk 1.66.0, but maybe we're missing something obvious that would allow us to use the latest versions of both Paperclip and aws-sdk.

In particular, https://github.com/thoughtbot/paperclip/blob/v4.3.5/lib/paperclip/storage/s3.rb contains many references to `AWS`, not `Aws` which results in the error: uninitialized constant Paperclip::Storage::S3::AWS.

**Comment by cllns** [comment]
If you want to stay on aws-sdk v2, you can get `master`, which will be released as 5.0
```
gem "paperclip", github: "thoughtbot/paperclip", ref: "c38bb31"
```

**Comment by tute** [comment]
In https://github.com/thoughtbot/paperclip/tree/v4.3.5#understanding-storage we specify aws 1, but we don't specify more to the top of the README. Can docs be improved for v4?

**Comment by warmwaffles** [comment]
Yes, this has been a source of frustration to see that 4.3.6 has been released, but in master it is at 4.3.1

**Comment by tute** [comment]
> Yes, this has been a source of frustration to see that 4.3.6 has been released, but in master it is at 4.3.1

@warmwaffles how exactly can we improve the README for v4.3 branch? master has now only mentions to `Aws` constant. Thank you!

**Comment by tute** [comment]
At the top of the README (https://github.com/thoughtbot/paperclip#documentation-valid-for-master-branch) we specify "Documentation valid for `master` branch" ("Please check the documentation for the paperclip version you are using: https://github.com/thoughtbot/paperclip/releases").

If this continues to be a source of confusion we'll add a special subsection in that piece of documentation.

Thanks!
